### PR TITLE
add twitter embed support

### PR DESCRIPTION
**Pull request by bjesus**
This PR uses the Managed Components Embed API to support server side generated embeds of a Twitter post.

For example, the following HTML:
```html
  <h3>another quote</h3>
  <div data-component-embed="twitter-post" data-tweet-id="1585735971198603264"></div>
```

will be server-side rendered like this:
screenshot_2023-03-14_14-42-37_216262485

### CODE REVIEW

**Comment by HammyHavoc** [comment]
Is this also compatible with the Twitter 'follow' button?

**Comment by bjesus** [comment]
You mean the "Follow" link next the to the user handle? it works, yes

**Comment by HammyHavoc** [comment]
This specifically: https://developer.twitter.com/en/docs/twitter-for-websites/follow-button/overview

I (regretfully) use these on most websites along with Twitter tweet embeds.

**Comment by bjesus** [comment]
I'm not sure I understand the usage of this Follow button - for me it just opens a new tab in which I can choose to follow a user. You can use any normal link to do this (e.g. https://twitter.com/intent/follow?screen_name=NASA ). Is the special thing about the button just the styling of it? Or is the JS doing something I a normal link can't do?

**Comment by HammyHavoc** [comment]
The button is able to display the follower count from a profile.

File: src/index.ts [hunk 1]
```diff
@@ -77,7 +85,67 @@ const onEvent =
...
+      async () => {
+        const res = await fetch(
+          `https://cdn.syndication.twimg.com/tweet-result?id=${tweetId}&token=${randomToken}`
+        )
+        return await res.json()
+      },
+      600 // Cache the Tweet for 10 minutes
+    )
+
+    const { user } = tweetResponse
```
Comment (Refaerds) [inline src/index.ts]:
```suggestion
    const { user, created_at } = tweetResponse
```

File: src/index.ts [hunk 2]
```diff
@@ -77,7 +85,67 @@ const onEvent =
...
+      dt.getFullYear()
+
+    let profileImage = manager.get('profileImage_' + user.screen_name)
+
+    if (!profileImage) {
+      const res = await fetch(user.profile_image_url_https)
+      profileImage = base64Encode(await res.arrayBuffer())
+      manager.set('profileImage_' + user.screen_name, profileImage)
+    }
+    console.log('this is text: ', tweetResponse.text)
```
Comment (Refaerds) [inline src/index.ts]: stray log

File: src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    { value: 1e3, symbol: 'k' },
+    { value: 1e6, symbol: 'M' },
+    { value: 1e9, symbol: 'G' },
+    { value: 1e12, symbol: 'T' },
+    { value: 1e15, symbol: 'P' },
+    { value: 1e18, symbol: 'E' },
+  ]
+  const rx = /\.0+$|(\.[0-9]*[1-9])0+$/
+  const item = lookup
+    .slice()
```
Comment (Refaerds) [inline src/utils.ts:13]: I think this .slice() can be removed

File: src/utils.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    .reverse()
```
Comment (Refaerds) [inline src/utils.ts:14]: why not just reverse the elements in lookup constant and avoid doing the .reverse()?

**Pull request merged**